### ISSUE TITLE
certain formatters return `\n` when there are no errors

### ISSUE THREAD

**Issue by alan-agius4** [palantir/tslint#4244]
### Bug Report

- __TSLint version__: 5.11.x

#### Actual behavior
When using the API

Ex: 
```ts
const Formatter = tslint.findFormatter('prose');
const formatter = new Formatter();
const output = formatter.format(result.failures, result.fixes);
```

Sometimes it will return ` \n` when no having an failures, or the failures have been fixed. This applies almost to all formatters.

See: https://github.com/palantir/tslint/blob/master/src/formatters/proseFormatter.ts#L34 and 
https://github.com/palantir/tslint/blob/master/src/formatters/proseFormatter.ts#L62

#### Expected behavior

I'd like that this return an empty string so we can test easily for a falsy value and don't emit anything to the console.

I'll happily do a PR to fix these issues, if you agree.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
